File: euler_010.py
from time import time
import primesieve

pgen = primesieve.Iterator()
psum = 0

start = time()
p = pgen.next_prime()
while p < 2000000:
    psum += p
    p = pgen.next_prime()

print("The sum of primes: " + str(psum))

stop = time()
print("Time: " + str(stop-start))

File: euler_010.js
const start = Date.now();

const upperBound = 2000000;
let sieve = [...Array(upperBound).keys()].fill(true);
let sum = 0;

for (let i = 2; i < upperBound; i++) {
  if (sieve[i]) {
    sum += i;
    let j = 2;
    while (i * j < upperBound) {
      sieve[i * j] = false;
      j++;
    }
  }
}

const end = Date.now();

console.log("Execute time: ", end - start, " Sum of primes is: ", sum);
